test/clj/rems/api/test_end_to_end.clj

```diff
@@ -607,3 +609,97 @@
           (testing "and it doesn't get rejected"
             (is (= "application.state/rejected" (:application/state (api-call :get (str "/api/applications/" app-3) nil
                                                                               api-key applicant-id))))))))))
+
+(deftest test-bonafide-bot
+  (let [api-key "42"
+        owner-id "owner"
+        applicant-id "bonafide-applicant"
+        applicant-attributes {:userid applicant-id
+                              :name "Bona Fide Applicant"
+                              :email "applicant@example.com"}
+        referer-id "bonafide-referer"
+        referer-attributes {:userid referer-id
+                            :name "Bona Fide Referer"
+                            :email "referer@example.com"
+                            :researcher-status-by "so"}
+        bot-attributes {:userid bonafide-bot/bot-userid
+                        :email nil
+                        :name "bonafide bot"}]
+    (testing "create users"
+      (api-call :post "/api/users/create" applicant-attributes api-key owner-id)
+      (api-call :post "/api/users/create" referer-attributes api-key owner-id)
+      (api-call :post "/api/users/create" bot-attributes api-key owner-id))
+    (let [resource-id
+          (extract-id
```

Comment:
I would prefer to have these on the same line. It does not change indentation much but at least it reserves the first column for names and second for values.

In `cond` we at least add blanks after each "section" and here that is both followed and not.

Reply:
I guess I forgot to add the blank lines, this style would definitely need them. This same two-line-let style is used in the other end-to-end tests and it makes it easier to see the repetition between the statements, especially in `test-end-to-end` which creates a lot of stuff. In this case everything is a bit simpler so I can switch to the standard let style.